packages/frontend/src/model/settings.ts

```diff
@@ -109,22 +141,11 @@ export const useSetSetting = (): useSetSettingReturn => {
         const apiAuthClient = makeApiAuthClient();
         const response = await apiAuthClient.patch(url, reqData(setting));
 
-        const updatedSetting = response.data as Setting;
-        const settingIndex = findIndex(
-          allSettings,
-          (s: Setting) => s._id === updatedSetting._id
-        );
-
-        if (settingIndex === -1) {
-          set(AllSettings, [updatedSetting]);
-        } else {
-          const updatedAllSettings = allSettings.slice();
-          updatedAllSettings.splice(settingIndex, 1, updatedSetting);
-
-          set(AllSettings, updatedAllSettings);
+        if (response.data) {
+          set(SingleSetting(setting._id), response.data);
         }
 
-        toast.success(`Saved setting "${updatedSetting.label}"`);
+        toast.success(`Saved setting "${response.data.label}"`);
```

Comment:
Should this be moved in the conditional above? I.e. only display the success notice if `response.data` is truthy and the local setting is updated?